range_sensor_layer/src/range_sensor_layer.cpp

```diff
@@ -376,7 +396,20 @@ void RangeSensorLayer::update_cell(double ox, double oy, double ot, double r, do
     ROS_DEBUG("%f %f | %f %f = %f", dx, dy, theta, phi, sensor);
     ROS_DEBUG("%f | %f %f | %f", prior, prob_occ, prob_not, new_prob);
     unsigned char c = to_cost(new_prob);
+
     setCost(x, y, c);
+    if(use_decay_)
+    {
+      if(c > to_cost(mark_threshold_))
+        point_map[std::make_pair(x, y)] = last_reading_time_.toSec();
```

Comment:
Since you use the pair(x,y) in all branches of this, can you declare it once inside the `use_decay` if statement?